src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -569,13 +569,19 @@ private BoundStatement BindLocalFunctionStatement(LocalFunctionStatementSyntax n
             {
                 expressionBody = runAnalysis(BindExpressionBodyAsBlock(node.ExpressionBody, diagnostics), diagnostics);
             }
-            else
+            else if (!hasErrors && (!localSymbol.IsExtern || !localSymbol.IsStatic))
             {
                 hasErrors = true;
                 diagnostics.Add(ErrorCode.ERR_LocalFunctionMissingBody, localSymbol.Locations[0], localSymbol);
```

Comment:
>ERR_LocalFunctionMissingBody [](start = 42, length = 28)

Instead of reporting this error for non-static extern local function, should we report a special error saying something like: "An extern local function must be static" ?

Reply:
The diagnostic message is "Local function '{0}' must either have a body or be marked 'static extern'."

Compare with the message when a method does not have a body: "'{0}' must declare a body because it is not marked abstract, extern, or partial"

---
In reply to: [354059428](https://github.com/dotnet/roslyn/pull/40011#discussion_r354059428) [](ancestors = 354059428)

Reply:
I will just modify the message for local functions to more closely resemble the one for methods unless you strongly believe another diagnostic is warranted.

---
In reply to: [355066096](https://github.com/dotnet/roslyn/pull/40011#discussion_r355066096) [](ancestors = 355066096,354059428)